pkg/scalers/kafka_scaler_test.go

```diff
@@ -21,17 +21,23 @@ var validMetadata = map[string]string{
 	"topic":         "my-topic",
 }
 
+// A complete valid authParams example for sasl, with username and passwd
+var validAuthParams = map[string]string{
+	"authMode": "sasl_plaintext",
+	"username": "admin",
+	"passwd":   "admin",
+}
+
 var parseKafkaMetadataTestDataset = []parseKafkaMetadataTestData{
 	{map[string]string{}, true, 0, nil, "", ""},
 	{map[string]string{"brokerList": "foobar:9092"}, true, 1, []string{"foobar:9092"}, "", ""},
 	{map[string]string{"brokerList": "foo:9092,bar:9092"}, true, 2, []string{"foo:9092", "bar:9092"}, "", ""},
 	{map[string]string{"brokerList": "a", "consumerGroup": "my-group"}, true, 1, []string{"a"}, "my-group", ""},
```

Comment:
I think the deleted test would still be valid and should pass.  May need to have two Test methods for `withAuthParams` and without auth params.  Not sure how you want to structure but my thought is to be non-breaking and use intelligent defaults the new auth params should be optional and should have a test for both flavors.

Reply:
@jeffhollan  you are right. i will add withAuthParams test.

Reply:
Now, the code should be merged. docs is done.